### ISSUE TITLE
Small hop to get into ratio with keyboard nav may hit ratio twice

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/ratio-and-proportion#354]
**Test device**
Lovelace and Dell
**Operating System**
ChromeOS and Win 10
**Browser**
Chrome
**Problem description**
For https://github.com/phetsims/QA/issues/609. 
When using keyboard nav with shift, if the hand may jump over ratio, it will instead make a smaller jump to be in ratio. In some cases it seems like this small jump is done in one direction, but not the other, and you can hit in ratio twice in that direction. It seems buggy, and @zepumph agreed over zoom, but I'm not certain it is worth addressing if it is a bug.
**Steps to reproduce**
1. Drag one hand so that is is off of any tickmarks
2. Use keyboard nav with shift to move the other hand over the in ratio portion
3. May be seen better with the log parameter

**Visuals**
![twicejumpin](https://user-images.githubusercontent.com/41024075/108275386-8004c880-7133-11eb-91e8-f6f1ce5c0ba2.gif)

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Ratio and Proportion‬
URL: https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-rc.2/phet/ratio-and-proportion_all_phet.html?log
Version: 1.0.0-rc.2 2021-02-13 05:25:44 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.150 Safari/537.36
Language: en-US
Window: 1280x658
Pixel Ratio: 1.5/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by KatieWoe** [comment]
Forgot to add screenshot provided by @zepumph 
![image](https://user-images.githubusercontent.com/41024075/108282602-299d8700-713f-11eb-9e27-4e193ad16f40.png)

**Comment by zepumph** [comment]
Thanks for the bug report @KatieWoe. I fixed this by ignoring the input snapping feature if we are already in Proportion. Will you please test on master this buggy case as well as this feature in general. Basically you should still always be able to access an in-proportion state when stepping with the shift-key, no matter the target ratio. It is most beneficial on the create screen with proportions like 7:10.

**Comment by zepumph** [comment]
Sorry. I think it will be easier to just test in the next RC.

**Comment by KatieWoe** [comment]
Looks fixed in rc.3

**Comment by KatieWoe** [comment]
I'm actually a bit unconfident with this one. @zepumph if this looks alright, please close. I think it is acceptable but I find it hard to be sure.
![unsure](https://user-images.githubusercontent.com/41024075/110694495-e9617f80-81a5-11eb-8117-f336ad01b365.gif)

**Comment by zepumph** [comment]
Yes that is really helpful. Thanks for that other case. I added both of these cases to unit tests, and that helped me find a more robust fix I think. I'll add to confirm in the next RC.

**Comment by zepumph** [comment]
I tried to cherry-pick, but got a merge conflict. Perhaps after I merge some of the other work this will be easier.

**Comment by zepumph** [comment]
cherry picked

**Comment by zepumph** [comment]
I'll double check this with in the RC to make sure things look good. @KatieWoe will do the same, but didn't feel quite as confident about making sure the actual algorithm seems fixed in the branch.

**Comment by KatieWoe** [comment]
Honestly, something still feels odd to me, but I once again am not sure how to put my finger on it. @zepumph if you want to zoom I might be able to show you.
![stillodd](https://user-images.githubusercontent.com/41024075/112206906-18cbaf80-8bdc-11eb-9b3f-a81b63dd4ab6.gif)

**Comment by zepumph** [comment]
Thanks for the video. It likely feels odd because the algorithm is odd! The gif above though is working as expected. Thanks for reporting it. Nothing looks wrong to me in that video or on my own side. I'm going to close this issue now.
